### ISSUE TITLE
Why does the same picture have a different inferenec result every time? why are the results so different from the same graph?

### ISSUE THREAD

**Issue by xu19971109** [open-mmlab/mmsegmentation#1771]
I have a similar question. I need to calculate the IOU of a single picture when inference pictures, but I find that the same picture runs repeatedly with the same model,  the results are actually different! Then I tried to use **np.unique (result[0])** for statistics, in my binary task, I found that the result was sometimes **0** and sometimes **[0,1]**, although flip was set to false by default. But why are the results so different from the same graph?

The following results are all from the **same model** and the **same image**
This is the IOU calculation I wrote myself
![image](https://user-images.githubusercontent.com/50237122/178888085-fb109cf6-6dd9-4180-97fc-7c26921a718e.png)
![image](https://user-images.githubusercontent.com/50237122/178888135-ce3da194-df43-4c9f-80d7-ed1d65b14166.png)
![image](https://user-images.githubusercontent.com/50237122/178889368-598e47bc-67cf-4344-8a5c-592a4f0e9307.png)

This is the IOU calculation function I use in mmseg
![image](https://user-images.githubusercontent.com/50237122/178888675-824f2ed4-103c-4ead-bcab-39c65e916ddf.png)
![image](https://user-images.githubusercontent.com/50237122/178888694-f9a8cfa9-73f7-4bf6-86cd-090801a6ae16.png)
![image](https://user-images.githubusercontent.com/50237122/178888773-d885f0dc-7460-496f-ab76-a51b3b59601e.png)

The iOU value is the same, which means that **result = inference_segmentor (model, img)** returns a different result each time

_Originally posted by @xu19971109 in https://github.com/open-mmlab/mmsegmentation/issues/1767#issuecomment-1183932718_

**Comment by xu19971109** [comment]
The dataset config is  a copy of Cityscape.
![image](https://user-images.githubusercontent.com/50237122/178890718-a7c41c4e-c047-425e-939f-11c5ee3e625c.png)

**Comment by xu19971109** [comment]
After my constant debugging, I found that the problem came from here：
![image](https://user-images.githubusercontent.com/50237122/178972233-c929218e-3f54-448b-8c22-9c0a06457711.png)
why？

**Comment by xu19971109** [comment]
![image](https://user-images.githubusercontent.com/50237122/178975739-d75366dd-d459-48e9-9457-597752bfd4a3.png)

**Comment by xu19971109** [comment]
I’find。
![image](https://user-images.githubusercontent.com/50237122/178976766-d922ccd0-2182-494a-a21c-3d9b47b47cb1.png)
I forgot to change the number of categories when I trained. The configuration file I trained was 26 categories, even though I trained with 0 and 1 labels. But when I corrected the number of categories during testing, it caused a different prediction. **Is this going to be a forced interrupt? And not just a print?** Because I've found that when num_classes is set wrong and only predicted once, it usually gets very close to the true value. But the second time I run it, it's a big gap
. Hopefully this will be updated in future releases.
